### PR TITLE
Replace PodFitsPorts with PodFitsHostPorts

### PR DESCRIPTION
**Pull request by HaiyangDING**
This PR addresses #14586

@davidopp @feihujiang

### CODE REVIEW

**Comment by feihujiang** [comment]
Change  doc as follows too.
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/scheduler_algorithm.md

**Comment by HaiyangDING** [comment]
@feihujiang 

The doc has been changed accordingly, thx for reminding this:)

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra @kubernetes/rh-ux @smarterclayton

File: plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func init() {
 func defaultPredicates() sets.String {
 	return sets.NewString(
 		// Fit is defined based on the absence of port conflicts.
-		factory.RegisterFitPredicate("PodFitsPorts", predicates.PodFitsPorts),
+		factory.RegisterFitPredicate("PodFitsHostPorts", predicates.PodFitsHostPorts),
```
Comment (liggitt) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: need to continue registering under the old name for backwards compatibility

File: plugin/pkg/scheduler/algorithmprovider/defaults/compatibility_test.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
   "predicates": [
     {"name": "MatchNodeSelector"},
     {"name": "PodFitsResources"},
-    {"name": "PodFitsPorts"},
+    {"name": "PodFitsHostPorts"},
```
Comment (liggitt) [inline plugin/pkg/scheduler/algorithmprovider/defaults/compatibility_test.go]: please read the comment on the test before changing it:
```
// Do not change this JSON. A failure indicates backwards compatibility with 1.0 was broken.
```

Comment (liggitt) [inline plugin/pkg/scheduler/algorithmprovider/defaults/compatibility_test.go]: if we want to rename it, that's fine, but the old name needs to stay registered as well. add a new item in the 1.1 test JSON below with any new predicate names

Comment (HaiyangDING) [inline plugin/pkg/scheduler/algorithmprovider/defaults/compatibility_test.go]: @liggitt 

Good point, I will handle the backwards compatibility issue in the next update.

**Comment by feihujiang** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by davidopp** [comment]
LGTM

Manually merging so I can do the cherrypick now.

**Comment by mikedanese** [comment]
Backwards compatibility issue was not addressed.

**Comment by liggitt** [comment]
I think it was. The old name was registered and the old JSON for the scheduler successfully deserializes, right?

**Comment by mikedanese** [comment]
I take it back, you are correct

**Pull request merged**